pkg/api/controllers/workspace/create.go

```diff
@@ -31,6 +32,12 @@ func CreateWorkspace(ctx *gin.Context) {
 		return
 	}
 
+	pattern := regexp.MustCompile(`^[a-zA-Z0-9._-]+$`).MatchString
+	if !pattern(createWorkspaceReq.Name) {
```

Comment:
This checks the workspace name (which is fine) but the issue was with the project names. (`createWorkspaceReq.Projects` ... `.Name`)

This check already exists in `WorkspaceService.CreateWorkspace` so I'd remove it from here.

![image](https://github.com/daytonaio/daytona/assets/25279767/4801370e-d450-4b0f-a810-b3b370f836af)

The checks should go to `pkg/server/workspaces/create.go` in the `createProject` function and return an error when the value isn't alphanumeric (plus hyphens, we can take care of the term later) - same as this line from `CreateWorkspace`: `isAlphaNumeric := regexp.MustCompile(^[a-zA-Z0-9-]+$).MatchString`